src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -189,6 +190,19 @@ private static Expression[] CreateArguments(ParameterInfo[]? parameters, Factory
                 args[i] = CreateArgument(parameters[i], factoryContext);
             }
 
+            if(factoryContext.HasAnotherBodyParameter)
```

Comment:
Nit: is it sufficient to call this `HasMultipleBodyParameters` because all we care about is the 1 or many cases?